### DISCUSSION TITLE
something happened and changed the values of some fields in one of my collection in milvus

### DISCUSSION THREAD

**Discussion by MaLo94** [milvus-io/milvus#22243] [Q&A and General discussion]
I had a collection with each data row having pk, guid, category, and embedding fields. Without making any changes in the code and changing the structure of the previous data, today the pk, guid, and category fields suddenly changed and have different values compared to before. What could be the reason for this?
values before this issue: 
![before](https://user-images.githubusercontent.com/119395346/219392445-07f57ba2-21ca-412c-ade9-da26c6c1dc2b.png)
values after this issue:
![after](https://user-images.githubusercontent.com/119395346/219392588-ddbffa82-5533-480a-ba2a-49e2185921ae.png)

**Comment by yhmo** [comment]
Could you provide more details about how you compare the data?

1. create a collection with 4 fields: "id", "post_id", "pk", "guide"    (which field is the primary key field? the "id"?)
2. how many rows were inserted into the collection?
3. call query() to retrieve the data to compare?

**Comment by MaLo94** [comment]
Sure
The fields i used to store my dataset in a milvus collection are ("Object_id", "Post_id", "Pk", "Embedding", "Category"). the primary key of each record is object_id that is a string guid (kind of string for speeding up in search).
about 200,000 rows I inserted into the collection and I did it with a kind of periodic insertion, I mean that I have made a timer function that inserts data after a period of time defined in batch format. 
I faced with a problem when I checked my recommender system that returned nothing because it did not find the object_id that I sent to the model to find the embedding and then similar records to the input. After that to recheck the system, I investigated Attu to see what is happening in my collections and wondered why the values of some fields changed! And now the main problem is I lost lots of data because of sudden changing happened.

**Reply by yhmo** [reply]
We need to do a test according to your description.

**Reply by xiaofan-luan** [reply]
what is the milvus release version?

**Reply by MaLo94** [reply]
> what is the milvus release version?

I installed milvus and other related tools to it with a docker-compose file, and get the images of milvus to use it. Here are the versions of each section: 
milvus-etcd --> 3.5.0
milvus-minio: RELEASE.2022-03-17T06-34-49z
milvus-standalone --> milvus: v2.2.0

**Comment by yhmo** [comment]
I write a script to verify the query() interface.
1. create a collection, with 4 fields "Object_id"(varchar), "Post_id"(varchar), "pk"(Int64), "Embedding"(float vector)
2. generate 200000 items, and store the items in a local txt file
3. insert the 200000 items into the collection
4. read the text file, fetch items one by one, use the query() interface to retrieve from milvus, compare the "Post_id" filed and "pk" filed

All these items are correct, no problem.

```
import random
import uuid
import os

from pymilvus import (
    connections,
    FieldSchema, CollectionSchema, DataType,
    Collection,
    utility
)


_HOST = '127.0.0.1'
_PORT = '19530'

# Const names
_COLLECTION_NAME = 'demo'

# Vector parameters
_DIM = 128

# Index parameters
_METRIC_TYPE = 'L2'
_INDEX_TYPE = 'IVF_FLAT'
_NLIST = 1024


# Create a Milvus connection
def create_connection():
    print(f"\nCreate connection...")
    connections.connect(host=_HOST, port=_PORT)
    print(f"\nList connections:")
    print(connections.list_connections())


def write_items(file_path, items:[]):
    print("write into", file_path)
    with open(file_path, "a") as file:
        for item in items:
            line = [item["Object_id"], item["Post_id"], str(item["pk"]), "\n"]
            token = "|"
            ss = token.join(line)
            file.write(ss)


def insert_items(collection, items:[]):
    cnt = len(items)
    data = [
        [items[i]["Object_id"] for i in range(cnt)],
        [items[i]["Post_id"] for i in range(cnt)],
        [items[i]["pk"] for i in range(cnt)],
        [items[i]["Embedding"] for i in range(cnt)],
    ]
    collection.insert(data)
    print(cnt, "items inserted")


def gen_data():
    if utility.has_collection(_COLLECTION_NAME):
        collection = Collection(_COLLECTION_NAME)
        collection.drop()

    field1 = FieldSchema(name="Object_id", dtype=DataType.VARCHAR, max_length=512, is_primary=True)
    field2 = FieldSchema(name="Post_id", dtype=DataType.VARCHAR, max_length=512)
    field3 = FieldSchema(name="pk", dtype=DataType.INT64)
    field4 = FieldSchema(name="Embedding", dtype=DataType.FLOAT_VECTOR, dim=_DIM)
    schema = CollectionSchema(fields=[field1, field2, field3, field4])
    collection = Collection(name=_COLLECTION_NAME, data=None, schema=schema)
    print("\ncollection created:", _COLLECTION_NAME)

    index_param = {
        "index_type": _INDEX_TYPE,
        "params": {"nlist": _NLIST},
        "metric_type": _METRIC_TYPE}
    collection.create_index("Embedding", index_param)

    file_path = "./items.txt"
    if os.path.exists(file_path):
        os.remove(file_path)

    items = []
    batch_cnt = 10000
    for i in range(200000):
        item = {
            "Object_id": str(uuid.uuid1()),
            "Post_id": str(uuid.uuid1()),
            "pk": i,
            "Embedding": [random.random() for _ in range(_DIM)],
        }
        items.append(item)
        if len(items) >= batch_cnt:
            write_items(file_path, items)
            insert_items(collection, items)
            items = []

    if len(items) > 0:
        write_items(file_path, items)
        insert_items(collection, items)

    collection.flush()
    print("wait for index finish...")
    utility.wait_for_index_building_complete(collection_name=_COLLECTION_NAME)


def verify_data():
    collection = Collection(name=_COLLECTION_NAME)
    collection.load()
    file_path = "./items.txt"
    with open(file_path) as file:
        token = file.readline()
        while token:
            if len(token) == 0:
                continue
            line = token.split(sep='|')
            token = file.readline()

            item = {
                "Object_id": line[0],
                "Post_id": line[1],
                "pk": line[2],
            }
            print("=======================================================================")
            print("original item:", item)

            query_expr = "Object_id in [\"{:s}\"]".format(line[0])
            print("query expression:", query_expr)
            result = collection.query(expr=query_expr, output_fields=["Post_id", "pk"])
            if len(result) > 0:
                print(f"result: {result[0]}")
                if not result[0]["Post_id"] == item["Post_id"]:
                    print(result[0]["Post_id"], "vs", item["Post_id"])
                    print("Not equal!!")
                    exit(1)
                if not str(result[0]["pk"]) == item["pk"]:
                    print(result[0]["pk"], "vs", item["pk"])
                    print("Not equal!!")
                    exit(1)

if __name__ == '__main__':
    create_connection()

    gen_data()
    verify_data()


```

**Comment by yhmo** [comment]
@MaLo94 
The Attu application doesn't display all the items on the "Data Preview" page to users. It randomly picks some items to display when you click the refresh button(I mark out it with a red circle).:
So, **your data is not changed**, you saw different items each time the page is refreshed.
![Screenshot from 2023-02-27 17-23-52](https://user-images.githubusercontent.com/2282099/221525197-a9a8148d-1442-4ebb-9131-6f450cdd9b49.png)

**Reply by MaLo94** [reply]
Thanks a lot for your rechecking and complete response to my issue. I know samples in the `Data Preview Tab` are random and by refreshing, I can get different items. About your demo code to check the issue I mentioned, I investigated it and compared it with my service in inserting and indexing in Milvus. I should say that almost everything is the same as you, Although after a while all the records in my Milvus collections changed, and I still can not find out why this has happened. Maybe other services in my production made these changes to the records, but I think there is no problem with them, cause if they made changes, they should do a query on my collection, fetch the data, change the values, and finally make a new collection. I still don't know how this happened to my collection suddenly. 

Again thanks a lot for your help
